auth/src/main/java/org/aerogear/mobile/auth/credentials/JwksManager.java

```diff
@@ -0,0 +1,184 @@
+package org.aerogear.mobile.auth.credentials;
+
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.aerogear.mobile.auth.Callback;
+import org.aerogear.mobile.auth.configuration.AuthServiceConfiguration;
+import org.aerogear.mobile.auth.configuration.KeycloakConfiguration;
+import org.aerogear.mobile.core.MobileCore;
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+import org.aerogear.mobile.core.http.HttpServiceModule;
+import org.aerogear.mobile.core.logging.Logger;
+import org.jose4j.jwk.JsonWebKeySet;
+import org.jose4j.lang.JoseException;
+
+import java.util.Date;
+
+import static org.aerogear.mobile.core.utils.SanityCheck.nonNull;
+
+/**
+ * A class that is responsible for manage the Json Web Key Set(JWKS).
+ */
+public class JwksManager {
+
+    private static final int MILLISECONDS_PER_MINUTE = 60*1000;
+
+    private static final String STORE_NAME = "org.aerogear.mobile.auth.JwksStore";
+
+    private static final String ENTRY_SUFFIX_FOR_KEY_CONTENT = "jwks_content";
+    private static final String ENTRY_SUFFIX_FOR_REQUEST_DATE = "requested_date";
+
+    private final HttpServiceModule httpModule;
+    private final AuthServiceConfiguration authServiceConfiguration;
+    private final SharedPreferences sharedPrefs;
+    private static final Logger logger = MobileCore.getLogger();
+
+    public JwksManager(@NonNull final Context context,
+                       @NonNull final MobileCore mobileCore,
+                       @NonNull final AuthServiceConfiguration authServiceConfiguration) {
+        this.httpModule = nonNull(mobileCore, "mobileCore").getHttpLayer();
```

Comment:
should we have another `nonNull` check on the httpModule as well?

Reply:
it would be reduntant: getHttpLayer can't return null - https://github.com/ziccardi/aerogear-android-sdk/blob/d258418c93b74eaf5600a7bda4afc0f83b65192e/core/src/main/java/org/aerogear/mobile/core/MobileCore.java#L72-L72